src/providers/MeetingProvider/MeetingManager.ts

```diff
@@ -671,10 +664,25 @@ export class MeetingManager implements AudioVideoObserver {
         this.selectVideoInputDeviceError = error;
         console.error('Failed to choose video input device.', error);
       }
-      this.selectedVideoInputDevice = deviceId;
+      let innerDevice = null;
+      if (isVideoTransformDevice(device)) {
+        innerDevice = await device.intrinsicDevice();
+      } else {
+        innerDevice = device;
+      }
+
+      const deviceId =
+        DefaultDeviceController.getIntrinsicDeviceId(innerDevice);
+      if (typeof deviceId === 'string') {
+        this.selectedVideoInputDevice = deviceId;
+      } else if (Array.isArray(deviceId) && deviceId[0]) {
+        this.selectedAudioInputDevice = deviceId[0];
+      }
+      this.selectedVideoInputTransformDevice = device;
     }
     this.publishSelectedVideoInputDevice();
     this.publishSelectVideoInputDeviceError();
+    this.publishSelectedVideoInputTransformDevice();
```

Comment:
Should we publish the video input transform device selection only when the device is a video transform device say inside line 668, inside `if (isVideoTransformDevice(device))` check?

Reply:
In V3, we want to combine line 684 and 685 - we want to have only one state that we publish. The problem is that the current selectedVideoInputDevice only holds a deviceId and not a VideoTransformDevice.
Instead of checking between selectedVideoInputDevice and selectedVideoInputTransformDevice every time, we will just use selectedVideoInputTransform device for the Device or VideoTransformDevice scenario.